src/l500/l500-color.cpp

```diff
@@ -221,7 +230,70 @@ namespace librealsense
                 return intrinsics;
             }
         }
-        throw std::runtime_error( to_string() << "intrinsics for resolution " << profile.width << "," << profile.height << " don't exist" );
+        throw std::runtime_error( to_string() << "intrinsics for resolution " << width << ","
+                                              << height << " don't exist" );
+    }
+
+    double l500_color_sensor::read_temperature() const
+    {
+        auto hwm = *_owner->_hw_monitor;
```

Comment:
In the function in `ac_trigger`, it uses a `weak_ptr` which can get a null if the device went away. Make sure this cannot crash please.

Reply:
done